File: quicksort.py
list_one = [99, 2, 32, 4, 1, 5, 8, 31]
list_two = [99, 2, 32]
def quicksort(numbers): # list of numbers to sort
    if len(numbers) == 0 or len(numbers) == 1:
        return numbers
    elif len(numbers) == 2:
        if numbers[0] > numbers[1]:
            numbers[0], numbers[1] = numbers[1], numbers[0]
        return numbers 
    else:
        pivot = numbers[0]
        s_nums = []
        g_nums = []

        for num in numbers:
            if num < pivot:
                s_nums.append(num)
            elif num > pivot: 
                g_nums.append(num)        
        s_nums.append(pivot)
        return quicksort(s_nums) + quicksort(g_nums) 

print "Program Start" 
print quicksort(list_two)
print quicksort(list_one) 
        

File: quicksort.go
package main

import "fmt"

func main() {
    list_one := []float64{6, 2, 32, 4, 1, 5, 8, 31}
    fmt.Println("Slice to sort:", list_one)
    fmt.Println(quicksort(list_one))


}

func quicksort(numbers []float64) []float64 {
    if len(numbers) == 0 || len(numbers) == 1 {
        return numbers
    }

    if len(numbers) == 2 {
        if numbers[0] > numbers[1] {
            numbers[0], numbers[1] = numbers[1], numbers[0]
        }
        return numbers
    }

    pivot := numbers[0]
    s_nums := make([]float64, 0, len(numbers))
    g_nums := make([]float64, 0, len(numbers))

    for _, num := range(numbers) {
        if num < pivot {
            s_nums = append(s_nums, num)
        }

        if num > pivot {
            g_nums = append(g_nums, num)
        }

    }
    
    s_nums = append(s_nums, pivot)
    
    
    return append(quicksort(s_nums), quicksort(g_nums)...)
}


